compiler/describe/analyze.go

```diff
@@ -28,13 +31,15 @@ type (
 		Meta string `json:"meta"`
 	}
 	Pool struct {
-		Kind string      `json:"kind"`
-		Name string      `json:"name"`
-		ID   ksuid.KSUID `json:"id"`
+		Kind     string      `json:"kind"`
+		Name     string      `json:"name"`
+		ID       ksuid.KSUID `json:"id"`
+		Inferred bool        `json:"inferred"`
 	}
 	Path struct {
-		Kind string `json:"kind"`
-		URI  string `json:"uri"`
+		Kind     string `json:"kind"`
+		URI      string `json:"uri"`
+		Inferred bool   `json:"inferred"`
```

Comment:
I mean technically you'll never use this endpoint for non-pools but I left that in for completion and because you might want to use this package to inspect source of a different kind. But to your point path can be inferred. At least on the semantic side if no source exists we prepend `&dag.DefaultScan{Kind: "DefaultScan"}` to the sequence.